### ISSUE TITLE
feature request: supporting t3s in copydomain?

### ISSUE THREAD

**Issue by ltutar** [biemond/biemond-orawls#350]
Hi Edwin,

When I look at the logging of managed server, I see that t3 is used instead of t3s:
```
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[unpack wls_domain]/returns: >>  succeed: write Domain to "/opt/wls/wlsdomains/domains/wls_domain"
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[unpack wls_domain]/returns: << close template
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[unpack wls_domain]/returns: >>  succeed: close template
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[unpack wls_domain]/returns: executed successfully
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Yaml_setting[domain wls_domain]/ensure: created
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/File[enroll.py wls_domain wls_domain]/ensure: defined content as '{md5}40e73e1c97dfe0ddd84a79a954c276af'
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: 
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: CLASSPATH=/opt/wls/middleware11g/patch_wls1036/profiles/default/sys_manifest_classpath/weblogic_patch.jar:/opt/wls/middleware11g/patch_ocp371/profiles/default/sys_manifest_classpath/weblogic_patch.jar:/usr/java/jdk1.7.0_80/lib/tools.jar:/opt/wls/middleware11g/wlserver_10.3/server/lib/weblogic_sp.jar:/opt/wls/middleware11g/wlserver_10.3/server/lib/weblogic.jar:/opt/wls/middleware11g/modules/features/weblogic.server.modules_10.3.6.0.jar:/opt/wls/middleware11g/wlserver_10.3/server/lib/webservices.jar:/opt/wls/middleware11g/modules/org.apache.ant_1.7.1/lib/ant-all.jar:/opt/wls/middleware11g/modules/net.sf.antcontrib_1.1.0.0_1-0b2/lib/ant-contrib.jar::/opt/wls/middleware11g/utils/config/10.3/config-launch.jar::/opt/wls/middleware11g/wlserver_10.3/common/derby/lib/derbynet.jar:/opt/wls/middleware11g/wlserver_10.3/common/derby/lib/derbyclient.jar:/opt/wls/middleware11g/wlserver_10.3/common/derby/lib/derbytools.jar::
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: 
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: Initializing WebLogic Scripting Tool (WLST) ...
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: 
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: Jython scans all the jar files it can find at first startup. Depending on the system, this process may take a few minutes to complete, and WLST may not return a prompt right away.
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: 
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: Welcome to WebLogic Server Administration Scripting Shell
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: 
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: Type help() for help on available commands
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: 
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: Connecting to t3://192.168.234.95:7001 with userid weblogic ...
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: Successfully connected to Admin Server 'AdminServer' that belongs to domain 'wls_domain'.
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: 
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: Warning: An insecure protocol was used to connect to the 
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: server. To ensure on-the-wire security, the SSL port or 
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: Admin port should be used instead.
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: 
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: Enrolling this machine with the domain directory at /opt/wls/wlsdomains/domains/wls_domain ...
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: Successfully enrolled this machine with the domain directory at /opt/wls/wlsdomains/domains/wls_domain.
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[execwlst wls_domain wls_domain]/returns: executed successfully
Notice: /Stage[main]/Profiles::Wls::Copy_domain/Orawls::Copydomain[wls_domain]/Exec[domain.py wls_domain wls_domain]/returns: executed successfully
Notice: Applied catalog in 373.20 seconds
[root@wlsagent1 ~]# 
[root@wlsagent1 ~]# 

```
and when I look at copydomain, I can not see where I can define the jsse and the rest.
Is my assumption true that copydomain does not support t3s? 
https://github.com/biemond/biemond-orawls/blob/master/manifests/copydomain.pp#L211

**Comment by ltutar** [comment]
my yaml for managed server 1 (different machine than admin server):
```
---
orautils::node_mgr_address:        "%{hiera('node1_address')}"

copy_domain_instances:
  "%{hiera('wls_domain_name')}":
    use_ssh:              false
    domain_pack_dir:      '/tmp'
    log_output:           true

nodemanager_instances:
  'nodemanager':
    nodemanager_address:                   "%{hiera('node1_address')}"
    nodemanager_port:                      "%{hiera('node_manager_port')}"
    domain_name:                           "%{hiera('wls_domain_name')}"
    nodemanager_secure_listener:           true 
    custom_identity:                       true
    custom_identity_keystore_filename:     "%{hiera('custom_identity_keystore_directory')}/%{hiera('node1_fqdn')}/identity_%{hiera('node1_fqdn')}.jks"
    custom_identity_keystore_passphrase:   "%{hiera('custom_identity_keystore_passphrase')}"
    custom_identity_alias:                 "%{hiera('node1_fqdn')}" 
    custom_identity_privatekey_passphrase: "%{hiera('custom_identity_privatekey_passphrase')}"
    log_output:                            true
```

**Comment by biemond** [comment]
indeed, it doesn't support it.

probably I need to change the erb template and set the trust parameters

**Comment by ltutar** [comment]
ok. thank you. After this change, I can turn of HTTP protocol of Admin Server.

**Comment by biemond** [comment]
yeah will do this also in the next few days

**Comment by biemond** [comment]
Fixed in  61af37a

**Comment by ltutar** [comment]
#362 solved the last error. I will close this. Thank you.
